### PR TITLE
don't reject sshsecret if it does not have a username

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/origin/issues/18742

@openshift/sig-developer-experience fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
note reviewers, a few lines below the code change there is a `fixNull(username)` call that deals with the fact that the optional username param is null

**Comment by gabemontero** [comment]
[merge]

File: src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java [hunk 1]
```diff
@@ -274,10 +273,9 @@ private static Credentials secretToCredentials(Secret secret) {
     private static Credentials newSSHUserCredential(String secretName,
             String username, String sshKeyData) {
         if (secretName == null || secretName.length() == 0 ||
-                username == null || username.length()== 0 ||
                 sshKeyData == null || sshKeyData.length() == 0) {
             logger.log(Level.WARNING, "Invalid secret data, secretName: " +
-                secretName + " username: " + username + " sshKeyData: " +
+                secretName + " sshKeyData: " +
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: i wonder if logging the secret key values is a bad idea from a security perspective...

Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: yeah probably ... aborting merge, will update

**Comment by gabemontero** [comment]
sensitive data no longer logged

File: src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java [hunk 2]
```diff
@@ -295,8 +293,10 @@ private static Credentials newUsernamePasswordCredentials(
                 usernameData == null || usernameData.length()== 0 ||
                         passwordData == null || passwordData.length() == 0) {
             logger.log(Level.WARNING, "Invalid secret data, secretName: " +
-                secretName + " usernameData: " + usernameData + " passwordData: " +
-                passwordData);
+                secretName + " usernameData is null: " + (usernameData == null)
+                + " usernameData is empty: " + (usernameData.length() == 0) + 
+                " passwordData is null: " + (passwordData == null) + 
+                " passwordData is empty: " + (passwordData.length() == 0));
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: won't this throw an NPE if passwordData==null?

Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: no the passwordData == null check just prior prevents that

once any of those conditions are true it goes in the if block and skips any additional conditionals

Comment (bparees) [reply src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: this isn't an if block, it's a print statement.

**Comment by gabemontero** [comment]
On Mon, Feb 26, 2018 at 2:33 PM, Ben Parees <notifications@github.com>
wrote:

> *@bparees* commented on this pull request.
> ------------------------------
>
> In src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java
> <https://github.com/openshift/jenkins-sync-plugin/pull/210#discussion_r170708912>
> :
>
> > @@ -295,8 +293,10 @@ private static Credentials newUsernamePasswordCredentials(
>                  usernameData == null || usernameData.length()== 0 ||
>                          passwordData == null || passwordData.length() == 0) {
>              logger.log(Level.WARNING, "Invalid secret data, secretName: " +
> -                secretName + " usernameData: " + usernameData + " passwordData: " +
> -                passwordData);
> +                secretName + " usernameData is null: " + (usernameData == null)
> +                + " usernameData is empty: " + (usernameData.length() == 0) +
> +                " passwordData is null: " + (passwordData == null) +
> +                " passwordData is empty: " + (passwordData.length() == 0));
>
> this isn't an if block, it's a print statement.
>
man i'm multitasking too much  ... yep

**Comment by gabemontero** [comment]
ok update to logs to account for null strings pushed

File: src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java [hunk 3]
```diff
@@ -274,11 +273,11 @@ private static Credentials secretToCredentials(Secret secret) {
     private static Credentials newSSHUserCredential(String secretName,
             String username, String sshKeyData) {
         if (secretName == null || secretName.length() == 0 ||
-                username == null || username.length()== 0 ||
                 sshKeyData == null || sshKeyData.length() == 0) {
             logger.log(Level.WARNING, "Invalid secret data, secretName: " +
-                secretName + " username: " + username + " sshKeyData: " +
-                    sshKeyData);
+                secretName + " sshKeyData is null: " + (sshKeyData == null) +
+                " sskKeyData is empty: " +
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: ssk->ssh

**Comment by bparees** [comment]
one more nit and lgtm

File: src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java (hunk 3 above)
Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java]: cool thanks ... update in a moment

**Comment by gabemontero** [comment]
[merge]

**Pull request merged**